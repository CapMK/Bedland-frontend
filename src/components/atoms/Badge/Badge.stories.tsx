import type { Meta, StoryObj } from '@storybook/react';
import GlobalStyle from 'theme/globalStyles';
import ThemeProvider from 'theme/ThemeContext';
import Badge from './Badge';
import { BADGE_VARIANTS } from './Badge.utils';

const meta = {
  title: 'Atoms/Badge',
  component: Badge,
  tags: ['autodocs'],
  argTypes: {
    variant: {
      options: BADGE_VARIANTS,
      control: { type: 'radio' },
      description: 'Badge variant',
      defaultValue: { summary: 'primary' },
    },
  },
  decorators: [
    (Story) => (
      <ThemeProvider>
        <GlobalStyle />
        {Story()}
      </ThemeProvider>
    ),
  ],
} satisfies Meta<typeof Badge>;
export default meta;
type Story = StoryObj<typeof meta>;
export const Default: Story = {
  args: {
    variant: 'primary',
    children: 'Test',
  },
};
