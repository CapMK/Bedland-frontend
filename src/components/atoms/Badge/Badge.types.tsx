import { HexColor } from 'theme/theme.types';

export const BadgeVariants = ['primary', 'secondary', 'danger', 'warning'] as const;

export type BadgeVariantProps = (typeof BadgeVariants)[number];

export type BadgeVariantsType = { [key in BadgeVariantProps]: key };

export type BadgeProps = {
  variant: BadgeVariantProps;
};
export type StyledBadgePropses = {
  colors: {
    bgColor: HexColor;
    color: HexColor;
    borderColor: HexColor | 'transparent';
  };
};
