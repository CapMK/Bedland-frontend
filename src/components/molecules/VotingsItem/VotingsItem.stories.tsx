import type { Meta, StoryObj } from '@storybook/react';
import GlobalStyle from 'theme/globalStyles';
import ThemeProvider from 'theme/ThemeContext';
import VotingsItem from './VotingsItem';

const meta = {
  title: 'Molecules/VotingsItem',
  component: VotingsItem,
  tags: ['autodocs'],
  decorators: [
    (Story) => (
      <ThemeProvider>
        <GlobalStyle />
        {Story()}
      </ThemeProvider>
    ),
  ],
  argTypes: {
    deadlineDate: {
      control: 'date',
      description: 'Select date as deadline date to calculate left/ago days from current date',
    },
  },
} satisfies Meta<typeof VotingsItem>;

export default meta;
type Story = StoryObj<typeof meta>;

export const Default: Story = {
  args: {
    deadlineDate: new Date(2023, 2, 16),
  },
};
