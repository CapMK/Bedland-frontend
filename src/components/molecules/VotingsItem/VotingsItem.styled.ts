import styled from 'styled-components';
import { PaletteType } from 'theme/theme.types';

export const VotingsItemContainer = styled.div<{ palette: PaletteType }>`
  align-items: center;
  border-bottom: 1px solid ${({ palette }) => palette.sidebarTabs};
  display: flex;
  flex-direction: column;
  margin-left: 1.125rem;
  margin-right: 1.125rem;
  width: calc(100% - 2.25rem);
  :last-child {
    margin-bottom: 0.938rem;
  }
`;

export const VotingsItemHeader = styled(VotingsItemContainer)`
  border-bottom: none;
  cursor: pointer;
  flex-direction: row;
  padding: 2rem 0 2rem 0rem;
  :last-child {
    margin-bottom: 0rem;
  }
`;
export const UnfoldContainer = styled.div<{ isUnfold: boolean; date: number }>`
  display: flex;
  flex-direction: column;
  gap: 1.875rem;
  height: ${({ isUnfold, date }) => {
    if (isUnfold) {
      return date < 0 ? '20rem' : '10rem';
    }
    return '0px';
  }};
  justify-content: flex-start;
  opacity: ${({ isUnfold }) => (isUnfold ? 1 : 0)};
  overflow: hidden;
  transition: opacity 0.25s linear, height 0.3s linear, visibility 0.3s linear;
  visibility: ${({ isUnfold }) => (isUnfold ? 'visible' : 'hidden')};
  width: calc(100% - 2.25rem);
`;

export const ChartWrapper = styled.div`
  height: 12.5rem;
  width: 27.25rem;
`;

export const IconWrapper = styled.div<{ isUnfold: boolean }>`
  transform: ${({ isUnfold }) => (isUnfold ? 'rotate(270deg)' : 'rotate(90deg)')};
  transition-duration: 0.3s;
`;

export default VotingsItemContainer;
