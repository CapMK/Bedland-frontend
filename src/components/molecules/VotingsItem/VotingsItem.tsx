import Icon from 'components/atoms/Icon/Icon';
import Typography from 'components/atoms/Typography/Typography';
import { useContext, useState } from 'react';
import { mockDataForVotingChart } from 'utils/mock/chartsMockData';
import { ThemeContext } from 'theme/ThemeContext';
import Button from 'components/atoms/Button/Button';
import CustomChart from '../CustomChart/CustomChart';
import * as Styled from './VotingsItem.styled';
import { dateDiffinDays, addLeadingZero, switchVariant } from './VotingsItem.utils';

export type VotingsItemProps = {
  deadlineDate: Date;
};

const VotingsItem = ({ deadlineDate }: VotingsItemProps) => {
  const { palette } = useContext(ThemeContext);
  const [isUnfold, setIsUnfold] = useState<boolean>(false);
  const newDeadlineDate = new Date(deadlineDate);
  const curDate = new Date();

  const dataForm = `${addLeadingZero(
    newDeadlineDate.getMonth() + 1,
  )}/${newDeadlineDate.getFullYear()}`;

  return (
    <Styled.VotingsItemContainer palette={palette}>
      <Styled.VotingsItemHeader
        palette={palette}
        onClick={() => setIsUnfold(!isUnfold)}
        data-testid='votings-item'
      >
        <Icon name='document' color='inputGrey' size={20} isActive={false} />
        <Typography
          variant='header4'
          style={{
            marginLeft: '1.188rem',
            flex: 1,
            fontWeight: dateDiffinDays(curDate, newDeadlineDate) < 0 ? '400' : '600',
          }}
          color='inputGrey'
        >
          Resolution {dataForm}
        </Typography>
        <Typography
          variant='header5'
          style={{
            marginRight: '1.438rem',
          }}
        >
          {switchVariant(curDate, newDeadlineDate)}
        </Typography>
        <Styled.IconWrapper isUnfold={isUnfold} data-testid='votings-item-icon'>
          <Button
            variant='icon'
            icon='arrowRight'
            iconSize={20}
            iconColor='inputGrey'
            onClick={() => null}
          />
        </Styled.IconWrapper>
      </Styled.VotingsItemHeader>

      <Styled.UnfoldContainer
        isUnfold={isUnfold}
        date={dateDiffinDays(curDate, newDeadlineDate)}
        data-testid='unfolded-container'
      >
        <Typography
          variant='dataInputsAndTooltips'
          color='inputGrey'
          style={{ padding: '10px 6px' }}
        >
          Resolution on the transfer of the water supply network located in the housing estate to
          the municipal water supply and sewage company
        </Typography>

        {dateDiffinDays(curDate, newDeadlineDate) < 0 ? (
          <Styled.ChartWrapper>
            <CustomChart variant='voting' data={mockDataForVotingChart} title='voting example' />
          </Styled.ChartWrapper>
        ) : (
          <Button wrapper style={{ display: 'flex', justifyContent: 'flex-end', width: '100%' }}>
            <Button wrapper style={{ marginRight: '2.8rem' }}>
              <Button variant='primary' type='button'>
                Vote now
              </Button>
            </Button>
          </Button>
        )}
      </Styled.UnfoldContainer>
    </Styled.VotingsItemContainer>
  );
};

export default VotingsItem;
