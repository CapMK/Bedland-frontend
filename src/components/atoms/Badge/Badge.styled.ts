import styled from 'styled-components';
import { StyledBadgePropses } from './Badge.types';

export const PrimaryBadge = styled.div<StyledBadgePropses>`
  align-items: center;
  background-color: ${({ colors: { bgColor } }) => bgColor};
  border: 1px solid ${({ colors: { borderColor } }) => borderColor};
  border-radius: 8px;
  color: ${({ colors: { color } }) => color};
  display: flex;
  height: 1.5rem;
  justify-content: center;
  padding: 0 1rem;
  width: 100%;
`;

export const SecondaryBadge = styled(PrimaryBadge)<StyledBadgePropses>`
  background-color: ${({ colors: { bgColor } }) => bgColor};
  border: 1px solid ${({ colors: { borderColor } }) => borderColor};
  color: ${({ colors: { color } }) => color};
`;

export const WarningBadge = styled(PrimaryBadge)<StyledBadgePropses>`
  background-color: ${({ colors: { bgColor } }) => bgColor};
  border: 1px solid ${({ colors: { borderColor } }) => borderColor};
  color: ${({ colors: { color } }) => color};
`;

export const DangerBadge = styled(PrimaryBadge)<StyledBadgePropses>`
  background-color: 1px solid ${({ colors: { bgColor } }) => bgColor};
  border: 1px solid ${({ colors: { borderColor } }) => borderColor};
  color: ${({ colors: { color } }) => color};
`;
