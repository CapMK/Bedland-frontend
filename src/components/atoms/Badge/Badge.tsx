import { FC, PropsWithChildren } from 'react';
import { BadgeProps } from './Badge.types';
import { getStyledBadge, BADGE_VARIANTS } from './Badge.utils';

const Badge: FC<PropsWithChildren<BadgeProps>> = ({
  children,
  variant = BADGE_VARIANTS.primary,
}) => {
  const { StyledBadge, colors } = getStyledBadge(variant);

  return <StyledBadge colors={colors}>{children}</StyledBadge>;
};

export default Badge;
