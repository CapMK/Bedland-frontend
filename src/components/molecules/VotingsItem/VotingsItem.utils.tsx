import Badge from 'components/atoms/Badge/Badge';

export const dateDiffinDays = (startDate: Date, endDate: Date) => {
  const utcEndDate = Date.UTC(endDate.getFullYear(), endDate.getMonth(), endDate.getDate());
  const utcStartDate = Date.UTC(startDate.getFullYear(), startDate.getMonth(), startDate.getDate());
  return Math.floor((utcEndDate - utcStartDate) / (24 * 60 * 60 * 1000));
};

export const addLeadingZero = (date: number) => {
  const stringDate = date.toString();
  return stringDate.length === 1 ? `0${stringDate}` : stringDate;
};

export const switchVariant = (curDate: Date, newDeadlineDate: Date) => {
  switch (true) {
    case dateDiffinDays(curDate, newDeadlineDate) < 0:
      return (
        <Badge variant='secondary'>
          {Math.abs(dateDiffinDays(curDate, newDeadlineDate))} days ago
        </Badge>
      );
    case dateDiffinDays(curDate, newDeadlineDate) < 8:
      return <Badge variant='danger'>{dateDiffinDays(curDate, newDeadlineDate)} days left</Badge>;
    case dateDiffinDays(curDate, newDeadlineDate) < 15:
      return <Badge variant='warning'>{dateDiffinDays(curDate, newDeadlineDate)} days left</Badge>;
    default:
      return <Badge variant='primary'>{dateDiffinDays(curDate, newDeadlineDate)} days left</Badge>;
  }
};
