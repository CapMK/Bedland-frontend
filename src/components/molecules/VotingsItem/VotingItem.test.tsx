import { renderWithProviders } from 'utils/test.utils';
import Badge from 'components/atoms/Badge/Badge';
import { fireEvent, screen, cleanup } from '@testing-library/react';
import VotingsItem, { VotingsItemProps } from './VotingsItem';
import { addLeadingZero, dateDiffinDays, switchVariant } from './VotingsItem.utils';

describe('VotingsItem', () => {
  afterEach(cleanup);
  const item = ({ deadlineDate }: VotingsItemProps) =>
    renderWithProviders(<VotingsItem deadlineDate={deadlineDate} />, {});

  it('should renders correctly VotingsItem', () => {
    expect(
      item({
        deadlineDate: new Date(2023, 4, 19),
      }),
    ).toMatchSnapshot();
  });
  it('dateDiffinDays should calculate correctly positive difference between two dates', () => {
    const result = dateDiffinDays(new Date(2023, 3, 19), new Date(2023, 3, 21));
    expect(result).toEqual(2);
  });
  it('dateDiffinDays should calculate correctly negative difference between two dates', () => {
    const result = dateDiffinDays(new Date(2023, 3, 21), new Date(2023, 3, 19));
    expect(result).toEqual(-2);
  });

  it('addLeadingZero should add 0 at the beginning when result is less than 10 ', () => {
    const result = addLeadingZero(new Date(2023, 3, 5).getMonth() + 1);
    expect(result).toEqual('04');
  });
  it('addLeadingZero shouldnt add 0 at the beginning when result is bigger than 10 ', () => {
    const result = addLeadingZero(new Date(2023, 10, 5).getMonth() + 1);
    expect(result).toEqual('11');
  });
  it('switch variant switches correctly variants', () => {
    const result = switchVariant(new Date(2023, 10, 5), new Date(2023, 10, 6));
    expect(result).toEqual(
      <Badge variant='danger'>
        {dateDiffinDays(new Date(2023, 10, 5), new Date(2023, 10, 6))} days left
      </Badge>,
    );
  });
  it('votings item with voting successfuly unfolds and arrow icon rotates', async () => {
    item({
      deadlineDate: new Date(2023, 5, 20),
    });
    const unfold = screen.getByTestId('votings-item');
    const votingsItemIcon = screen.getByTestId('votings-item-icon');
    const unfoldedContainer = screen.getByTestId('unfolded-container');
    expect(unfoldedContainer).toHaveStyle('height: 0');
    expect(votingsItemIcon).toHaveStyle('transform: rotate(90deg)');
    await fireEvent.click(unfold);
    expect(unfoldedContainer).toHaveStyle('height: 10rem');
    expect(votingsItemIcon).toHaveStyle('transform: rotate(270deg)');
  });
  it('votings item with chart successfuly unfolds and arrow icon rotates', async () => {
    item({
      deadlineDate: new Date(1995, 11, 17),
    });
    const unfold = screen.getByTestId('votings-item');
    const votingsItemIcon = screen.getByTestId('votings-item-icon');
    const unfoldedContainer = screen.getByTestId('unfolded-container');
    expect(unfoldedContainer).toHaveStyle('height: 0');
    expect(votingsItemIcon).toHaveStyle('transform: rotate(90deg)');
    await fireEvent.click(unfold);
    expect(unfoldedContainer).toHaveStyle('height: 20rem');
    expect(votingsItemIcon).toHaveStyle('transform: rotate(270deg)');
  });
});
