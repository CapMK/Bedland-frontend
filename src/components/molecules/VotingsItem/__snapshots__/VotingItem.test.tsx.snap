// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`VotingsItem > should renders correctly VotingsItem 1`] = `
{
  "asFragment": [Function],
  "baseElement": .c2 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #30BA57;
  border: 1px solid transparent;
  border-radius: 8px;
  color: #fff;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 1.5rem;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 0 1rem;
  width: 100%;
}

.c6 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #F56513;
  border: 1px solid #F56513;
  border-radius: 0.5rem;
  color: #FCFEFF;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  gap: 10px;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 0.5625rem 1rem;
  white-space: nowrap;
  width: 100%;
  font-family: Source Sans Pro,sans-serif;
  font-size: 1rem;
  font-weight: 600;
  line-height: 1.4;
  -webkit-letter-spacing: 1px;
  -moz-letter-spacing: 1px;
  -ms-letter-spacing: 1px;
  letter-spacing: 1px;
}

.c6:hover {
  background-color: #FF7C33;
  border: 1px solid #FF7C33;
}

.c4 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: transparent;
  color: #4553DD;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-bottom: 1px solid #CACACA;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  margin-left: 1.125rem;
  margin-right: 1.125rem;
  width: calc(100% - 2.25rem);
}

.c0:last-child {
  margin-bottom: 0.938rem;
}

.c1 {
  border-bottom: none;
  cursor: pointer;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  padding: 2rem 0 2rem 0rem;
}

.c1:last-child {
  margin-bottom: 0rem;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  gap: 1.875rem;
  height: 0px;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  opacity: 0;
  overflow: hidden;
  -webkit-transition: opacity 0.25s linear,height 0.3s linear,visibility 0.3s linear;
  transition: opacity 0.25s linear,height 0.3s linear,visibility 0.3s linear;
  visibility: hidden;
  width: calc(100% - 2.25rem);
}

.c3 {
  -webkit-transform: rotate(90deg);
  -ms-transform: rotate(90deg);
  transform: rotate(90deg);
  -webkit-transition-duration: 0.3s;
  transition-duration: 0.3s;
}

<body>
    <div>
      <div
        class="c0"
      >
        <div
          class="c0 c1"
          data-testid="votings-item"
        >
          <svg
            data-testid="document-icon"
            fill="none"
            height="20"
            viewBox=" 0 0 20 19"
            width="20"
            xmlns="http://www.w3.org/2000/svg"
          >
            <mask
              id="mask-document"
            />
            <path
              d="M17 17.5H3C1.4 17.5 1 16.167 1 15.5V3.5C1 2.96957 1.21071 2.46086 1.58579 2.08579C1.96086 1.71071 2.46957 1.5 3 1.5H13C13.5304 1.5 14.0391 1.71071 14.4142 2.08579C14.7893 2.46086 15 2.96957 15 3.5V9.5M17 17.5C16.333 17.5 15 17.1 15 15.5V9.5M17 17.5C18.6 17.5 19 16.167 19 15.5V9.5H15"
              stroke="#4B4B4B"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="1.67"
            />
          </svg>
          <h4
            style="font-family: Source Sans Pro, sans-serif; font-size: 1rem; font-weight: 600; line-height: 1.4; letter-spacing: 1px; color: rgb(75, 75, 75); margin-left: 1.188rem; flex: 1;"
          >
            Resolution 
            05/2023
          </h4>
          <h5
            style="font-family: Source Sans Pro, sans-serif; font-weight: 600; font-size: 0.875rem; line-height: 1.4; letter-spacing: 0; margin-right: 1.438rem;"
          >
            <div
              class="c2"
            >
              22
               days left
            </div>
          </h5>
          <div
            class="c3"
            data-testid="votings-item-icon"
          >
            <button
              class="c4"
            >
              <svg
                data-testid="arrowRight-icon"
                fill="none"
                height="20"
                viewBox="0 0 20 20"
                width="20"
                xmlns="http://www.w3.org/2000/svg"
              >
                <mask
                  id="mask-arrowRight"
                />
                <path
                  d="M8.33325 5.83334L12.4999 10L8.33325 14.1667"
                  stroke="#4B4B4B"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="1.39167"
                />
              </svg>
            </button>
          </div>
        </div>
        <div
          class="c5"
          data-testid="unfolded-container"
        >
          <p
            style="font-family: Source Sans Pro, sans-serif; font-weight: 400; font-size: 0.875rem; line-height: 1.4; letter-spacing: 1px; color: rgb(75, 75, 75); padding: 10px 6px;"
          >
            Resolution on the transfer of the water supply network located in the housing estate to the municipal water supply and sewage company
          </p>
          <div
            class=""
            style="display: flex; justify-content: flex-end; width: 100%;"
          >
            <div
              class=""
              style="margin-right: 2.8rem;"
            >
              <button
                class="c6"
                type="button"
              >
                Vote now
              </button>
            </div>
          </div>
        </div>
      </div>
      <div
        data-testid="location-display"
      >
        /
      </div>
    </div>
  </body>,
  "container": .c2 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #30BA57;
  border: 1px solid transparent;
  border-radius: 8px;
  color: #fff;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 1.5rem;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 0 1rem;
  width: 100%;
}

.c6 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #F56513;
  border: 1px solid #F56513;
  border-radius: 0.5rem;
  color: #FCFEFF;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  gap: 10px;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 0.5625rem 1rem;
  white-space: nowrap;
  width: 100%;
  font-family: Source Sans Pro,sans-serif;
  font-size: 1rem;
  font-weight: 600;
  line-height: 1.4;
  -webkit-letter-spacing: 1px;
  -moz-letter-spacing: 1px;
  -ms-letter-spacing: 1px;
  letter-spacing: 1px;
}

.c6:hover {
  background-color: #FF7C33;
  border: 1px solid #FF7C33;
}

.c4 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: transparent;
  color: #4553DD;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-bottom: 1px solid #CACACA;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  margin-left: 1.125rem;
  margin-right: 1.125rem;
  width: calc(100% - 2.25rem);
}

.c0:last-child {
  margin-bottom: 0.938rem;
}

.c1 {
  border-bottom: none;
  cursor: pointer;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  padding: 2rem 0 2rem 0rem;
}

.c1:last-child {
  margin-bottom: 0rem;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  gap: 1.875rem;
  height: 0px;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  opacity: 0;
  overflow: hidden;
  -webkit-transition: opacity 0.25s linear,height 0.3s linear,visibility 0.3s linear;
  transition: opacity 0.25s linear,height 0.3s linear,visibility 0.3s linear;
  visibility: hidden;
  width: calc(100% - 2.25rem);
}

.c3 {
  -webkit-transform: rotate(90deg);
  -ms-transform: rotate(90deg);
  transform: rotate(90deg);
  -webkit-transition-duration: 0.3s;
  transition-duration: 0.3s;
}

<div>
    <div
      class="c0"
    >
      <div
        class="c0 c1"
        data-testid="votings-item"
      >
        <svg
          data-testid="document-icon"
          fill="none"
          height="20"
          viewBox=" 0 0 20 19"
          width="20"
          xmlns="http://www.w3.org/2000/svg"
        >
          <mask
            id="mask-document"
          />
          <path
            d="M17 17.5H3C1.4 17.5 1 16.167 1 15.5V3.5C1 2.96957 1.21071 2.46086 1.58579 2.08579C1.96086 1.71071 2.46957 1.5 3 1.5H13C13.5304 1.5 14.0391 1.71071 14.4142 2.08579C14.7893 2.46086 15 2.96957 15 3.5V9.5M17 17.5C16.333 17.5 15 17.1 15 15.5V9.5M17 17.5C18.6 17.5 19 16.167 19 15.5V9.5H15"
            stroke="#4B4B4B"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="1.67"
          />
        </svg>
        <h4
          style="font-family: Source Sans Pro, sans-serif; font-size: 1rem; font-weight: 600; line-height: 1.4; letter-spacing: 1px; color: rgb(75, 75, 75); margin-left: 1.188rem; flex: 1;"
        >
          Resolution 
          05/2023
        </h4>
        <h5
          style="font-family: Source Sans Pro, sans-serif; font-weight: 600; font-size: 0.875rem; line-height: 1.4; letter-spacing: 0; margin-right: 1.438rem;"
        >
          <div
            class="c2"
          >
            22
             days left
          </div>
        </h5>
        <div
          class="c3"
          data-testid="votings-item-icon"
        >
          <button
            class="c4"
          >
            <svg
              data-testid="arrowRight-icon"
              fill="none"
              height="20"
              viewBox="0 0 20 20"
              width="20"
              xmlns="http://www.w3.org/2000/svg"
            >
              <mask
                id="mask-arrowRight"
              />
              <path
                d="M8.33325 5.83334L12.4999 10L8.33325 14.1667"
                stroke="#4B4B4B"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="1.39167"
              />
            </svg>
          </button>
        </div>
      </div>
      <div
        class="c5"
        data-testid="unfolded-container"
      >
        <p
          style="font-family: Source Sans Pro, sans-serif; font-weight: 400; font-size: 0.875rem; line-height: 1.4; letter-spacing: 1px; color: rgb(75, 75, 75); padding: 10px 6px;"
        >
          Resolution on the transfer of the water supply network located in the housing estate to the municipal water supply and sewage company
        </p>
        <div
          class=""
          style="display: flex; justify-content: flex-end; width: 100%;"
        >
          <div
            class=""
            style="margin-right: 2.8rem;"
          >
            <button
              class="c6"
              type="button"
            >
              Vote now
            </button>
          </div>
        </div>
      </div>
    </div>
    <div
      data-testid="location-display"
    >
      /
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "store": {
    "@@observable": [Function],
    "dispatch": [Function],
    "getState": [Function],
    "replaceReducer": [Function],
    "subscribe": [Function],
  },
  "unmount": [Function],
}
`;
